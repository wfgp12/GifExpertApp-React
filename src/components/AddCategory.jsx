import { useState } from "react";
import PropTypes from 'prop-types';

// eslint-disable-next-line react/prop-types
export const AddCategory = ({ onSubmit }) => {
  const [inputValue, setInputValue] = useState('');

  const handleChange = (e) => {
    setInputValue(e.target.value)
  }

  const handleSubmit = (event) => {
    event.preventDefault()
    if (inputValue.trim().length <= 1) return;

    onSubmit(inputValue.trim());
    setInputValue('');
  }

  return (
    <form onSubmit={handleSubmit} aria-label="form">
      <input
        type="text"
        placeholder="buscar gifs"
        value={inputValue}
        onChange={handleChange}
      />
    </form>
  )
}


AddCategory.propTypes = {
  onSubmit: PropTypes.func.isRequired
}