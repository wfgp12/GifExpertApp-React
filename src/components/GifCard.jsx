
// eslint-disable-next-line react/prop-types
export const GifCard = ({ title, url }) => {
    return (
        <div className="card">
            <img src={url}/>
            <p>{title}</p>
        </div>
    )
}
