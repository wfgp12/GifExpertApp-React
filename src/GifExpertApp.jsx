import { useState } from "react"
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Dragon Ball']);

    const handleAddCategory = (newCat) => {
        if (categories.find(cat => cat.toLowerCase() === newCat.toLowerCase())) {
            const filterCat = categories.filter(cat => cat.toLowerCase() !== newCat.toLowerCase());
            setCategories([newCat, ...filterCat]);
        } else {
            setCategories(cat => [newCat, ...cat]);
        }
    }

    return (
        <>
            <h1>GifExpertApp</h1>

            <AddCategory onSubmit={handleAddCategory} />

            {categories.map((category, index) => 
                <GifGrid 
                    key={index} 
                    category={category} 
                />
            )}

        </>
    )
}
