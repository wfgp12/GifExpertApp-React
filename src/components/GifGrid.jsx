import PropTypes from 'prop-types';

import { useFetchGifs } from "../hooks";
import { GifCard } from "./GifCard";

export const GifGrid = ({ category }) => {
    const { gifs, isLoading } = useFetchGifs(category)


    return (
        <>
            <h3>{category}</h3>
            {isLoading &&<h2>Cargando...</h2>}
            <div className="card-grid">
                {gifs.map((image) =>
                    <GifCard
                        key={image.id}
                        {...image}
                    />
                )}
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}