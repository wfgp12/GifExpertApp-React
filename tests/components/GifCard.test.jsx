import { render, screen } from "@testing-library/react"
import { GifCard } from '../../src/components/'

/* eslint-disable no-undef */
describe('Pruebas realizadas al componente <GifCard />', () => {
    const title = 'League of Legends';
    const url = 'https://media3.giphy.com/media/h5C2kiESzaypGjXtTY/giphy-downsized-medium.gif?cid=30e1a8980bwmomcq8f4b3s8ykghycqqpca6srj9wp753clme&ep=v1_gifs_search&rid=giphy-downsized-medium.gif&ct=g';
    test('debe hace match con el snapshot', () => {
        const { container } = render(
            <GifCard 
                {...{
                    title,
                    url
                }}
            />
        );

        expect(container).toMatchSnapshot();
    })

    test('debe de mostrar la imagen con el URl y ALT indicado', () => { 
        render(<GifCard {...{title, url}}/>);

        expect(screen.getByRole('img').src).toBe(url);
        expect(screen.getByRole('img').alt).toBe(title);
     })
})