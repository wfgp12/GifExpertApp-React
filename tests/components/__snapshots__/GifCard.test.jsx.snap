// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas realizadas al componente <GifCard /> debe hace match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="League of Legends"
      src="https://media3.giphy.com/media/h5C2kiESzaypGjXtTY/giphy-downsized-medium.gif?cid=30e1a8980bwmomcq8f4b3s8ykghycqqpca6srj9wp753clme&ep=v1_gifs_search&rid=giphy-downsized-medium.gif&ct=g"
    />
    <p>
      League of Legends
    </p>
  </div>
</div>
`;
